src/parser/util.js

```diff
@@ -86,3 +87,25 @@ pp.unexpected = function (pos, messageOrType = "Unexpected token") {
   }
   this.raise(pos != null ? pos : this.state.start, messageOrType);
 };
+
+pp.expectPlugin = function (name) {
+  if (!this.hasPlugin(name)) {
+    const pluginInfo = plugins[name];
+    const pointers = [];
+    if (pluginInfo) {
+      if (pluginInfo.babelSyntaxPlugin) {
+        pointers.push(`babel-plugin-${pluginInfo.babelSyntaxPlugin}`);
+      }
+      if (pluginInfo.babelTransformPlugins) {
+        pointers.push(...pluginInfo.babelTransformPlugins.map(
+          (shortName) => `babel-plugin-${shortName}`
+        ));
+      }
+    }
+    const leadingPointersOr = pointers.length ? ` ${pointers.join(", ")} or ` : " ";
+
+    this.unexpected(null, `This experimental syntax requires the ${name} parser plugin. `
+      + `Check out${ leadingPointersOr }the parserOpts.${name} Babel option.`);
```

Comment:
We should change it so that Babel users will use the transform plugin, and babylon users the syntax, and for Babel users not to use both.